src/bitrot.py

```diff
@@ -291,6 +291,22 @@ def select_all_paths(self, cur):
             row = cur.fetchone()
         return result
 
+    def select_all_hashes(self, cur):
+        result = {}
+        cur.execute('SELECT hash, path FROM bitrot order by hash')
+        row = cur.fetchone()
+        chash = ''
+        while row: 
+            rhash, rpath = row
+            if chash != rhash:
```

Comment:
It's simpler if you rather do:
```
result.setdefault(rhash, set).add(rpath)
```

Then you don't need to order the SELECT which should be even faster.